src/parcsr_ls/par_mgr_device.c

```diff
@@ -664,7 +664,7 @@ hypre_ParCSRMatrixExtractBlockDiagDevice( hypre_ParCSRMatrix   *A,
     *-----------------------------------------------------------------*/
    {
       dim3 bDim = hypre_GetDefaultDeviceBlockDimension();
-      dim3 gDim = hypre_GetDefaultDeviceGridDimension(num_rows / blk_size, "warp", bDim);
+      dim3 gDim = hypre_GetDefaultDeviceGridDimension(num_blocks, "warp", bDim);
```

Comment:
Is num_blocks = num_rows / blk_size? Or has the argument here changed? num_blocks is set to num_points above.

Reply:
This was a bug :)

Correct is `num_blocks`